### PR TITLE
Issue/1362 Add Order Shipping Method

### LINKED ISSUE #1362: Order Detail: Add shipping method to the shipping details section
As discussed, it would be nice to add the shipping method of an order in the Order Detail and Order Fulfillment screen.

Ref. p99K0U-1IW-p2

### PR DESCRIPTION
**Pull request by pmusolino**
Fixes #1362. This PR adds the shipping method details to the Order detail and order Fulfill screen. 

## Changes
-  Adds the shipping method below the shipping address details in Order detail and order fulfill screen.
- The shipping card is only displayed for orders containing at least one virtual product. It should not be displayed for virtual products.
- There are plugins that allow multiple shipping methods for an order. It was decided that we would display a warning notice to the user, if an order contains multiple shipping methods. This warning notice will be displayed only on the Order Detail page.
- Only the first shipping method of an order will be displayed (if there are multiple shipping methods).

## Note
The shipping method text is localized with the language configured inside the store.

#### Physical product with single shipping methods in an order
(LEFT: Order Detail . RIGHT: Order Fulfill)
 . 

## Testing
- Order with only virtual products - should not display warning - should not display shipping method or warning notice.
- Order with physical + virtual products with single shipping methods - display the shipping method but the shipping warning should not be displayed.
- Order with physical + virtual products with multiple shipping methods - display the shipping method and the shipping warning notice.

Update release notes:

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/Classes/Extensions/UIImage+Woo.swift [hunk 1]
```diff
@@ -13,6 +13,13 @@ extension UIImage {
         return Gridicon.iconOfType(.addOutline)
     }
 
+    /// Notice Icon
+    ///
+    static var noticeImage: UIImage {
```
Comment (ctarda) [inline WooCommerce/Classes/Extensions/UIImage+Woo.swift]: We've been adding a simple unit test for each icon in `IconsTests`. When I started that, it made sense for icons that where UIImage(named:), to make sure that we did not ship with images missing from the Assets catalog (been bit by that in previous lifes).  I am not sure of the value of adding a test for GridIcons as well, but we've been doing it 🤷‍♂

File: WooCommerce/Classes/ViewModels/OrderDetailsViewModel/OrderDetailsDataSource.swift [hunk 2]
```diff
@@ -542,6 +608,8 @@ extension OrderDetailsDataSource {
                 return OrderTrackingTableViewCell.reuseIdentifier
             case .trackingAdd:
                 return LeftImageTableViewCell.reuseIdentifier
+            case .shippingNotice:
+                return LeftImageMultilineTableViewCell.reuseIdentifier
```
[the remark this answers is not in the document]
Comment (pmusolino) [reply WooCommerce/Classes/ViewModels/OrderDetailsViewModel/OrderDetailsDataSource.swift]: I didn't realize that this type of cell already existed. I would be inclined to unify. I await the opinion of others.

File: WooCommerce/Classes/Extensions/UIImage+Woo.swift (hunk 1 above)
Comment (pmusolino) [reply WooCommerce/Classes/Extensions/UIImage+Woo.swift]: You did well to report it. I added the test. ;)

File: WooCommerce/Classes/ViewRelated/Orders/Order Details/Product List Section/Fulfillment/FulfillViewController.swift [hunk 3]
```diff
@@ -585,11 +611,20 @@ private extension FulfillViewController {
 }
 
 
+// MARK: - Lookup orders and statuses
+private extension FulfillViewController {
+    func containOnlyVirtualProducts(for products: [Product]) -> Bool {
```
[the remark this answers is not in the document]
Comment (pmusolino) [reply WooCommerce/Classes/ViewRelated/Orders/Order Details/Product List Section/Fulfillment/FulfillViewController.swift]: No more needed, I removed it (useless)

File: WooCommerce/Classes/ViewModels/OrderDetailsViewModel/OrderDetailsDataSource.swift (hunk 2 above)
[the remark this answers is not in the document]
Comment (jaclync) [reply WooCommerce/Classes/ViewModels/OrderDetailsViewModel/OrderDetailsDataSource.swift]: Maybe we can keep both versions, and we can leave unifying the cells to a separate task with @kyleaparker 's thoughts?

File: WooCommerce/Classes/ViewModels/OrderDetailsViewModel/OrderDetailsDataSource.swift [hunk 4]
```diff
@@ -191,108 +209,124 @@ private extension OrderDetailsDataSource {
...
+    func configureShippingNotice(cell: LeftImageMultilineTableViewCell) {
+        let cellTextContent = NSLocalizedString(
+            "This order is using extensions to calculate shipping. The shipping methods shown might be incomplete.",
+            comment: "Shipping notice row label when there is more than one shipping method")
+        cell.leftImage = Icons.shippingNoticeIcon
+        cell.labelText = cellTextContent
+        cell.selectionStyle = .none
+        
+        cell.accessibilityTraits = .staticText
+        cell.accessibilityLabel = NSLocalizedString(
```
[the remark this answers is not in the document]
Comment (pmusolino) [reply WooCommerce/Classes/ViewModels/OrderDetailsViewModel/OrderDetailsDataSource.swift]: You're right. L243 needs to be `accessibilityHint`

File: WooCommerce/Classes/ViewModels/OrderDetailsViewModel/OrderDetailsDataSource.swift (hunk 2 above)
[the remark this answers is not in the document]
Comment (kyleaparker) [reply WooCommerce/Classes/ViewModels/OrderDetailsViewModel/OrderDetailsDataSource.swift]: Unifying sounds good 👍

**Comment by pmusolino** [comment]
> * The images on the fulfil screen look stretched

This was fixed in another PR https://github.com/woocommerce/woocommerce-ios/pull/1438

---- 
> * Across both screens the spacing between sections looks very inconsistent, we should have consistent spacing throughout the app

You are right. The problem is also present in the current release of the app. The right spacing seems to be the one used on the Fulfill screen. I'm going to open a new issue for that.

---- 
> * The first section on the order screen with the customer name and status shouldn't have any space above it

Like above, the problem is also present in the current release of the app. I'm going to open a new issue for that.

**Pull request merged**